app/src/main/java/org/simple/clinic/home/overdue/OverdueListAdapter.kt

```diff
@@ -28,138 +32,169 @@ class OverdueListAdapter @Inject constructor() : ListAdapter<OverdueListItem, Ov
-        }
+  override fun getItemViewType(position: Int) =
+      when (getItem(position)) {
+        is OverdueHeaderItem -> OVERDUE_HEADER
+        is OverduePatientItem -> OVERDUE_LIST
       }
-    }
 
-    return holder
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OverdueListViewHolder {
+    val layout = LayoutInflater.from(parent.context).inflate(viewType, parent, false)
+    return when (viewType) {
+      OVERDUE_HEADER -> HeaderViewHolder(layout)
+      else -> OverduePatientViewHolder(layout, itemClicks)
+    }
   }
 
   override fun onBindViewHolder(holder: OverdueListViewHolder, position: Int) {
-    holder.appointment = getItem(position)
-    holder.render()
+    if (holder is OverduePatientViewHolder) {
+      holder.appointment = getItem(position) as OverduePatientItem
+      holder.render()
+    }
   }
 }
 
-data class OverdueListItem(
-    val appointmentUuid: UUID,
-    val patientUuid: UUID,
-    val name: String,
-    val gender: Gender,
-    val age: Int,
-    val phoneNumber: String? = null,
-    val bpSystolic: Int,
-    val bpDiastolic: Int,
-    val bpDaysAgo: Int,
-    val overdueDays: Int,
-    val isAtHighRisk: Boolean
-)
-
-class OverdueListViewHolder(
-    itemView: View,
-    private val eventStream: PublishSubject<UiEvent>
-) : RecyclerView.ViewHolder(itemView) {
-
-  private val patientNameTextView by bindView<TextView>(R.id.overdue_patient_name_age)
-  private val patientBPTextView by bindView<TextView>(R.id.overdue_patient_bp)
-  private val overdueDaysTextView by bindView<TextView>(R.id.overdue_days)
-  private val isAtHighRiskTextView by bindView<TextView>(R.id.overdue_high_risk_label)
-  private val callButton by bindView<ImageButton>(R.id.overdue_patient_call)
-  private val actionsContainer by bindView<LinearLayout>(R.id.overdue_actions_container)
-  private val phoneNumberTextView by bindView<TextView>(R.id.overdue_patient_phone_number)
-  private val agreedToVisitTextView by bindView<TextView>(R.id.overdue_agreed_to_visit)
-  private val remindLaterTextView by bindView<TextView>(R.id.overdue_reminder_later)
-  private val removeFromListTextView by bindView<TextView>(R.id.overdue_remove_from_list)
-
-  lateinit var appointment: OverdueListItem
-
-  init {
-    callButton.setOnClickListener {
-      eventStream.onNext(CallPatientClicked(appointment.phoneNumber!!))
-    }
-    agreedToVisitTextView.setOnClickListener {
-      eventStream.onNext(AgreedToVisitClicked(appointment.appointmentUuid))
-    }
-    remindLaterTextView.setOnClickListener {
-      eventStream.onNext(RemindToCallLaterClicked(appointment.appointmentUuid))
-    }
-    removeFromListTextView.setOnClickListener {
-      eventStream.onNext(RemoveFromListClicked(appointment.appointmentUuid, appointment.patientUuid))
-    }
-  }
+sealed class OverdueListItem {
+
+  object OverdueHeaderItem : OverdueListItem()
+
+  data class OverduePatientItem(
```

Comment:
Considering that these are inner classes of `OverdueListItem`, prefixing them with `Overdue` is redundant. `OverdueListItem.Header` would be better than `OverdueListItem.OverdueHeaderItem`. Let's rename them?

Reply:
Done